src/index.ts

```diff
@@ -111,14 +118,27 @@ async function run() {
     };
 
     if (appService.defineToCreateOrUpdateApp(zendeskConfig) === 'UPDATE') {
-      const id = appId || (ids[env] as string);
+      const id = Number(appId || (ids[env] as string));
+
+      if (!Number.isInteger(id))
+        throw new Error(`Invalid appId. Expected a integer but got: ${id}`);
```

Comment:
Cara, aqui temos que validar de outras formas o id.
Como o id é uma sequência númerica, esse `Number(appId || (ids[env] as string))` faz a conversão para número corretamente, mas se por acaso um dos valores passado pra ele for `true` por exemplo, ele seta o id para `1`.

Talvez caia melhor uma validação após pegar o appId mas sem ter convertido ele para number ainda, só verificando se ele é uma string mesmo e após isso mudar para number para evitar validação com falso positivo.

Não sei precisa gastar tempo nisso não, mas uma ideia de uma validação mais completa (meramente exemplo)
```javascript
function validateAppId(appId) {
  if (!appId) return console.log("error not found");

  if (typeof appId !== "string" && typeof appId !== "number")
    return console.log("error type");
  
  if (!Number.isInteger(Number(appId))) return console.log("error invalid");

  console.log("valid");
}
```

Reply:
Muito Válido! Criei uma função genérica